plugins/outputs/zabbix/utils.go

```diff
@@ -0,0 +1,230 @@
+package zabbix
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/datadope-io/go-zabbix"
+	"github.com/influxdata/telegraf"
+)
+
+// processMetric converts a Telegraf metric to a list of Zabbix metrics.
+// Ignore metrics with no hostname.
+func (z Zabbix) processMetric(metric telegraf.Metric) []*zabbix.Metric {
+	hostname, ok := metric.Tags()["host"]
+	if !ok {
+		return nil
+	}
+
+	zbxMetrics := []*zabbix.Metric{}
+
+	for fieldName, value := range metric.Fields() {
+		zbxMetric, err := z.buildZabbixMetric(metric, hostname, fieldName, value)
+		if err != nil {
+			z.Log.Errorf("Error converting telegraf metric to Zabbix format: %v", err)
+			continue
+		}
+
+		zbxMetrics = append(zbxMetrics, zbxMetric)
+	}
+
+	return zbxMetrics
+}
+
+// buildZabbixMetric builds a Zabbix metric from a Telegraf metric, for one particular value.
+func (z Zabbix) buildZabbixMetric(metric telegraf.Metric, hostname, fieldName string, value interface{}) (*zabbix.Metric, error) {
+	metricValue, err := buildValue(value)
+	if err != nil {
+		return nil, fmt.Errorf("error converting value: %w", err)
+	}
+
+	key := fmt.Sprintf("%s%s.%s", z.Prefix, metric.Name(), fieldName)
+
+	if z.SkipMeasurementPrefix {
+		key = fmt.Sprintf("%s%s", z.Prefix, fieldName)
+	}
```

Comment:
How about simplifying this to
```suggestion
	key := z.Prefix + metric.Name() + "." +fieldName
	if z.SkipMeasurementPrefix {
		key = z.Prefix + fieldName
	}
```